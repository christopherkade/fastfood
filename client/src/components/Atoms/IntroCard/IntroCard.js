import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"

const CardWrapper = styled.div`
  display: flex;
  flex-direction: column;
  padding: 1rem;
  background: rgba(239,121,138, 0.9);
  text-align: center;
  border-radius: 4px;
`

const CardTitle = styled.h1`
  color: white;
  font-weight: 300;
  font-size: 3rem;
  margin: 0;
`

const CardSubtitle = styled.p`
  color: white;
  font-weight: 300;
  font-size: 2rem;
`

const StartButton = styled.button`
  border: 1px solid white;
  border-radius: 4px;
  background-color: transparent;
  color: white;
  font-size: 1.5rem;
  margin: 3.125rem 1rem 1rem 1rem;

  &:hover {
    cursor: pointer;
  }

  &:focus {
    outline-color: #E2DFE0;
  }
`

const IntroCard = ({ onClick }) => {
  return (
    <CardWrapper>
      <CardTitle>
        Fastfood
      </CardTitle>
      <CardSubtitle>
        Argue less, eat more
      </CardSubtitle>

      <StartButton onClick={onClick}>
        Let's go !
      </StartButton>
    </CardWrapper>
  )
}

IntroCard.propTypes = {
  onClick: PropTypes.func.isRequired,
};

export default IntroCard
