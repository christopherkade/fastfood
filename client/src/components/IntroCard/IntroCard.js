import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"

const CardWrapper = styled.div`
  display: flex;
  flex-direction: column;
  padding: 1rem;
  background-color: #EF798A;
  text-align: center;
  border-radius: 4px;
`

const CardTitle = styled.h1`
  color: white;
  font-weight: 300;
  font-size: 1.5rem;
  margin: 0;
`

const CardSubtitle = styled.p`
  color: white;
  font-weight: 300;
  font-size: 1rem;
`

const StartButton = styled.button`
  border: 1px solid white;
  border-radius: 4px;
  background-color: transparent;
  color: white;
  font-size: 1.5rem;
  margin: 3.125rem 1rem 1rem 1rem;

  &:hover {
    cursor: pointer;
  }

  &:focus {
    outline-color: #E2DFE0;
  }
`

const IntroCard = ({ onClick }) => {
  return (
    <CardWrapper>
      <CardTitle>
        Welcome to Fastfood
      </CardTitle>
      <span role="img" aria-label="">🍕🥑🍣🍜</span>
      <CardSubtitle>
        The easiest way to pick what to eat !
      </CardSubtitle>

      <StartButton onClick={onClick}>
        Start a session
      </StartButton>
    </CardWrapper>
  )
}

IntroCard.propTypes = {
  onClick: PropTypes.func.isRequired,
};

export default IntroCard
