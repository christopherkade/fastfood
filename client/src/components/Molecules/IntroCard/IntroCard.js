import React from "react"
import PropTypes from "prop-types"
import styled from "styled-components"

import { Title } from "components/Atoms/Title"

const CardWrapper = styled.div`
  display: flex;
  flex-direction: column;
  padding: 1rem;
  background: rgba(239,121,138, 0.9);
  text-align: center;
  border-radius: 4px;

  & > h1 {
    margin-bottom: 0;
  }
`

const CardSubtitle = styled.p`
  color: white;
  font-weight: 300;
  font-size: 2rem;
  margin: 0;
`

const CardTwitter = styled.p`
  color: white;
  margin: 0;
  font-size: 0.875rem;

  & > a {
    color: white;
    font-weight: bold;
    text-decoration: none;

    &:hover {
      opacity: 0.8;
    }
  }
`

const StartButton = styled.button`
  border: 1px solid white;
  border-radius: 4px;
  background-color: transparent;
  color: white;
  font-size: 1.5rem;
  margin: 3.125rem 1rem 1rem 1rem;

  &:hover {
    cursor: pointer;
    opacity: 0.8;
  }

  &:focus {
    outline-color: #E2DFE0;
  }
`

const IntroCard = ({ onClick }) => {
  return (
    <CardWrapper>
      <Title color="white" size="3">
        Fastfood
      </Title>

      <CardSubtitle>
        Argue less, eat more
      </CardSubtitle>

      <StartButton onClick={onClick}>
        Let's go !
      </StartButton>

      <CardTwitter>
        Made by <a href="http://twitter.com/christo_kade">@christo_kade</a>
      </CardTwitter>
    </CardWrapper>
  )
}

IntroCard.propTypes = {
  onClick: PropTypes.func.isRequired,
};

export default IntroCard
