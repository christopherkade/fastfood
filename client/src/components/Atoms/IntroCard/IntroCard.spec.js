import React from 'react'
import { render, cleanup } from 'react-testing-library'

import IntroCard from "./IntroCard"

afterEach(cleanup);

describe("<IntroCard />", () => {
  describe("Snapshots", () => {
    it("should match snapshot", () => {
      const { container } = render(<IntroCard onClick={() => { }} />);
      expect(container).toMatchSnapshot();
    });
  })
})
