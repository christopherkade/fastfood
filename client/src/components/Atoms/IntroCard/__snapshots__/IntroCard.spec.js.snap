// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IntroCard /> Snapshots should match snapshot 1`] = `
<div>
  <div
    class="sc-bdVaJa jOMnRk"
  >
    <h1
      class="sc-bwzfXH gyiady"
    >
      Fastfood
    </h1>
    <p
      class="sc-htpNat fnwEBe"
    >
      Argue less, eat more
    </p>
    <button
      class="sc-bxivhb vcWxY"
    >
      Let's go !
    </button>
  </div>
</div>
`;
